docs/ref/csp.txt

```diff
@@ -0,0 +1,260 @@
+header to responses. This allows the browser to monitor and report policy
+violations without blocking content, making it useful for refining a policy.
+
+Most browsers log CSP violations to the developer console and can optionally
+send them to a reporting endpoint. To collect these reports, the ``report-uri``
+directive must be defined (see :ref:`csp-reports` for more details).
+
+.. admonition:: These settings can be used independently or together
+
+   * Use :setting:`SECURE_CSP` alone to enforce a policy that has already been
+     tested and verified.
+   * Use :setting:`SECURE_CSP_REPORT_ONLY` on its own to evaluate a new policy
+     without disrupting site behavior. This mode does not block violations, it
+     only logs them. It's useful for testing and monitoring, but provides no
+     protection against active threats.
+   * Use *both* to maintain an enforced baseline while experimenting with
+     changes. Even for well-established policies, continuing to collect reports
+     reports can help detect regressions, unexpected changes in behavior, or
+     potential tampering in production environments.
+
+Settings structure
+------------------
+
+This example illustrates the settings structure::
+
+    from django.utils.csp import CSP
+
+    SECURE_CSP = {
+        "default-src": [CSP.SELF],
+        "img-src": ["data:", CSP.SELF, "https://images.example.com"],
+        "frame-src": [CSP.NONE],
+    }
+
+Both :setting:`SECURE_CSP` and :setting:`SECURE_CSP_REPORT_ONLY` settings must
+be dictionaries, where:
+
+* Keys are directive names, such as ``default-src`` or ``script-src``.
+* Values are lists, sets, or tuples of allowed sources or tokens. Sets are
+  automatically sorted to ensure consistent output in the generated headers.
+
+.. admonition:: Directives validation
+
+   Django's CSP middleware helps construct and send the appropriate header
+   based on your settings, but it does **not validate** that your directives
+   and values conform to the CSP specification. It is your responsibility to
+   ensure that your configuration is syntactically and semantically correct.
+   Use browser developer tools or external CSP validators during development.
+
+   See specific details on `available CSP directives
+   <https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy#directives>`_.
+
+.. _csp-reports:
+
+Policy violation reports
+========================
+
+When a CSP violation occurs, browsers typically log details to the developer
+console, providing immediate feedback during development. To also receive these
+reports programmatically, the policy must include a `reporting directive
+<https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Content-Security-Policy#reporting_directives>`_
+such as ``report-uri`` that specifies where violation data should be sent.
+
+Django supports configuring these directives via the relevant settings, but
```

Comment:
I think this section is not clear that this is quite `SECURE_CSP_REPORT_ONLY` specific information. For example, I think the "via the relevant settings" is "via the `SECURE_CSP_REPORT_ONLY` setting"